core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -4912,7 +4919,9 @@ class ReplicaManagerTest {
       assertTrue(fooPart eq fooPart2)
       val bar1 = new TopicPartition("bar", 1)
       replicaManager.markPartitionOffline(bar1)
-      assertEquals(None, replicaManager.getOrCreatePartition(bar1, emptyDelta, BAR_UUID))
+      val (barPart, barNew) = replicaManager.getOrCreatePartition(bar1, emptyDelta, BAR_UUID).get
+      assertTrue(barNew)
+      assertEquals(bar1, barPart.topicPartition)
```

Comment:
Could we add a test case for returning `None`?
Since after this PR, we'll have 2 cases when creating partition in OfflinePartition, we should test them both.

Reply:
Addressed in [cdf9c0f](https://github.com/apache/kafka/pull/15263/commits/cdf9c0f0cf817e68ca27092addb21308a9bc3762)